vmm/src/vm.rs

```diff
@@ -1001,12 +1007,23 @@ impl Vm {
             // If failed, retry to load it as UEFI binary.
             // As the UEFI binary is formatless, it must be the last option to try.
             Err(linux_loader::loader::Error::Pe(InvalidImageMagicNumber)) => {
+                let memory_slot = self.memory_manager.lock().unwrap().allocate_memory_slot();
+                let flash_memory = FlashMemory::new(
+                    &self.vm,
+                    arch::layout::UEFI_START,
+                    arch::layout::UEFI_SIZE as usize,
+                    memory_slot,
+                )
+                .unwrap();
+                let mem = flash_memory.get_memory();
```

Comment:
Since this is about flash memory, could you move most of the logic to the memory manager (adding the accessors you need to `MemoryManager`)?